File: Movie/API/MovieService.swift
//
//  MovieService.swift
//  Movie
//
//

import Foundation
import Alamofire

class MovieService {
    
    func getPopularMovies(page: Int) async throws -> MoviesResponse {
        return try await AF.request(
            "\(ApiConstants.baseUrl)/movie/popular?api_key=\(ApiConstants.apiKey)&page=\(page)"
        )
        .serializingDecodable(MoviesResponse.self, decoder: DateDecoder())
        .value
    }
    
    func getMovieDetail(id: Int) async throws -> MovieDetail {
        return try await AF.request(
            "\(ApiConstants.baseUrl)/movie/\(id)?api_key=\(ApiConstants.apiKey)"
        )
        .serializingDecodable(MovieDetail.self, decoder: DateDecoder())
        .value
    }
}

File: shared/src/commonMain/kotlin/com/mehmetaliocak/movie_app/api/MovieService.kt
package com.mehmetaliocak.movie_app.api

import com.mehmetaliocak.movie_app.models.MovieDetail
import com.mehmetaliocak.movie_app.models.MoviesResponse
import com.rickclephas.kmp.nativecoroutines.NativeCoroutines
import io.ktor.client.HttpClient
import io.ktor.client.call.body
import io.ktor.client.request.get
import org.koin.core.component.KoinComponent
import org.koin.core.component.inject

class MovieService : KoinComponent {

    private val client: HttpClient by inject()

    @NativeCoroutines
    suspend fun getPopularMovies(page: Int): MoviesResponse {
        return client.get("movie/popular") {
            url {
                parameters.append("page", "$page")
            }
        }.body()
    }

    @NativeCoroutines
    suspend fun getMovieDetail(id: Long): MovieDetail {
        return client.get("movie/$id").body()
    }
}